tests/integration/commands/functions-serve/functions-serve.test.ts

```diff
@@ -193,4 +202,127 @@ describe.concurrent('functions:serve command', () => {
       })
     })
   })
+
+  test('should inject AI Gateway when linked site and online', async (t) => {
+    await withSiteBuilder(t, async (builder) => {
+      const { siteInfo, aiGatewayToken, routes } = createAIGatewayTestData()
+      const checkFunction = createAIGatewayCheckFunction()
+
+      await builder
+        .withNetlifyToml({ config: { functions: { directory: 'functions' } } })
+        .withFunction({
+          path: checkFunction.path,
+          handler: () => {
```

Comment:
This is using the legacy Lambda syntax and I don't think the AI gateway is expected to work with the Lambda compatibility mode since we don't have a bootstrap there to populate the environment.

You can change the test to use a v2 function like this: https://github.com/netlify/cli/blob/961abeec8d354cf02699055d99baa15f9b6251a0/tests/integration/commands/dev/functions.test.ts#L141